crates/precompile/src/kzg_point_evaluation/blst.rs

```diff
@@ -0,0 +1,182 @@
+        return false;
+    };
+
+    // Get the trusted setup G2 point [τ]₂
+    let tau_g2 = get_trusted_setup_g2();
+
+    // Get generators
+    let g1 = get_g1_generator();
+    let g2 = get_g2_generator();
+
+    // Compute P_minus_y = commitment - [y]G₁
+    let y_g1 = p1_scalar_mul(&g1, &y_scalar);
+    let p_minus_y = p1_sub_affine(&commitment_point, &y_g1);
+
+    // Compute X_minus_z = [τ]G₂ - [z]G₂
+    let z_g2 = p2_scalar_mul(&g2, &z_scalar);
+    let x_minus_z = p2_sub_affine(&tau_g2, &z_g2);
+
+    // Verify: P - y = Q * (X - z)
+    // Using pairing check: e(P - y, -G₂) * e(proof, X - z) == 1
+    let neg_g2 = p2_neg(&g2);
+
+    pairing_check(&[(p_minus_y, neg_g2), (proof_point, x_minus_z)])
+}
+
+/// Get the trusted setup G2 point [τ]₂ from the Ethereum KZG ceremony.
+/// This is g2_monomial_1 from trusted_setup_4096.json
+fn get_trusted_setup_g2() -> blst_p2_affine {
+    // For compressed G2, we need to decompress
+    let mut g2_affine = blst_p2_affine::default();
+    unsafe {
+        // The compressed format has x coordinate and a flag bit for y
+        // We use deserialize_compressed which handles this automatically
+        let result = blst::blst_p2_deserialize(&mut g2_affine, TRUSTED_SETUP_TAU_G2_BYTES.as_ptr());
+        if result != blst::BLST_ERROR::BLST_SUCCESS {
+            panic!("Failed to deserialize trusted setup G2 point");
+        }
+    }
+    g2_affine
+}
+
+/// Get G1 generator point
+fn get_g1_generator() -> blst_p1_affine {
+    unsafe { ::blst::BLS12_381_G1 }
+}
+
+/// Get G2 generator point
+fn get_g2_generator() -> blst_p2_affine {
+    unsafe { ::blst::BLS12_381_G2 }
+}
+
+/// Parse a G1 point from compressed format (48 bytes)
+fn parse_g1_compressed(bytes: &[u8; 48]) -> Result<blst_p1_affine, PrecompileError> {
+    let mut point = blst_p1_affine::default();
+    unsafe {
+        let result = blst::blst_p1_deserialize(&mut point, bytes.as_ptr());
+        if result != blst::BLST_ERROR::BLST_SUCCESS {
+            return Err(PrecompileError::Other(
+                "Invalid compressed G1 point".to_string(),
+            ));
+        }
+
+        // Verify the point is on curve
+        if !blst_p1_affine_on_curve(&point) {
+            return Err(PrecompileError::Other("G1 point not on curve".to_string()));
+        }
+
+        // Verify the point is in the correct subgroup
+        if !blst_p1_affine_in_g1(&point) {
+            return Err(PrecompileError::Other(
+                "G1 point not in correct subgroup".to_string(),
+            ));
+        }
+    }
+    Ok(point)
+}
+
+/// Read a scalar field element from bytes and verify it's canonical
+fn read_scalar_canonical(bytes: &[u8; 32]) -> Result<blst_scalar, PrecompileError> {
+    let mut scalar = blst_scalar::default();
+
+    // Read scalar from big endian bytes
+    unsafe {
+        blst_scalar_from_bendian(&mut scalar, bytes.as_ptr());
+    }
+
+    if unsafe { !blst_scalar_fr_check(&scalar) } {
```

Comment:
have checked how c-kzg does it, and this is how it is done there. Using backward conversion with `blst_bendian_from_scalar` didn't pass tests